webv2/uniqueid/uniqueid_assignment.go

```diff
@@ -0,0 +1,219 @@
+
+							break
+						}
+					}
+
+				}
+
+				for sourceforeignkeyindex, sourceforeignkey := range sourcetable.ForeignKeys {
+
+					for spannerforeignkeyindex, spannerforeignkey := range spannertable.Fks {
+
+						if sourceforeignkey.Name == spannerforeignkey.Name {
+
+							foreignkeyid := GenerateForeignkeyId()
+
+							sourceforeignkey.Id = foreignkeyid
+							spannerforeignkey.Id = foreignkeyid
+
+							conv.SrcSchema[sourcetable.Name].ForeignKeys[sourceforeignkeyindex] = sourceforeignkey
+							conv.SpSchema[spannertable.Name].Fks[spannerforeignkeyindex] = spannerforeignkey
+						}
+
+					}
+				}
+
+				for sourcei, sourceindexes := range sourcetable.Indexes {
+
+					for spanneri, spannerindexes := range spannertable.Indexes {
+
+						if sourceindexes.Name == spannerindexes.Name {
+
+							indexesid := GenerateIndexesId()
+
+							sourceindexes.Id = indexesid
+							spannerindexes.Id = indexesid
+
+							conv.SrcSchema[sourcetable.Name].Indexes[sourcei] = sourceindexes
+							conv.SpSchema[spannertable.Name].Indexes[spanneri] = spannerindexes
+
+						}
+
+					}
+				}
+
+				updateSpannerTableIndexKeyOrder(spannertable)
+				updateSourceTableIndexKeyOrder(sourcetable)
+
+				conv.SrcSchema[sourcetablename] = sourcetable
+				conv.SpSchema[spannertablename] = spannertable
+				break
+			}
+		}
+	}
+
+}
+
+// updateSpannerTableIndexKeyOrder Update Primary Key Order as columnId.
+func updateSpannerTableIndexKeyOrder(spannertable ddl.CreateTable) {
+
+	for i := 0; i < len(spannertable.Pks); i++ {
+		for spannercolumnname, _ := range spannertable.ColDefs {
+			if spannertable.Pks[i].Col == spannercolumnname {
+
+				o := getSpannerColumnIndex(spannertable, spannercolumnname)
+				spannertable.Pks[i].Order = o
+
+			}
+		}
+	}
+}
+
+// updateSourceTableIndexKeyOrder Update Primary Key Order as columnId.
+func updateSourceTableIndexKeyOrder(sourcetable schema.Table) {
+
+	for i := 0; i < len(sourcetable.PrimaryKeys); i++ {
+		for sourcecolumnname, _ := range sourcetable.ColDefs {
+			if sourcetable.PrimaryKeys[i].Column == sourcecolumnname {
+
+				o := getSourceColumnIndex(sourcetable, sourcecolumnname)
+				sourcetable.PrimaryKeys[i].Order = o
+			}
+		}
+	}
+}
+
+// getSpannerColumnIndex return columnn index as Inserted Order.
+func getSpannerColumnIndex(spannertable ddl.CreateTable, columnName string) int {
+
+	for i := 0; i < len(spannertable.ColNames); i++ {
+		if spannertable.ColNames[i] == columnName {
+			return i + 1
+		}
+	}
+	return 0
+}
+
+// getColumnIndex return columnn index as Inserted Order.
+func getSourceColumnIndex(sourcetable schema.Table, columnName string) int {
+
+	for i := 0; i < len(sourcetable.ColNames); i++ {
+		if sourcetable.ColNames[i] == columnName {
+			return i + 1
+		}
+	}
+	return 0
+}
+
+func GenerateId() string {
+
+	sessionState := session.GetSessionState()
+
+	counter, _ := strconv.Atoi(sessionState.Counter.ObjectId)
```

Comment:
This seems like a round about way of just incrementing an integer and returning the string representation of it

Reply:
@bharadwaj-aditya yes